### PR TITLE
GOOFSEC 2: High Frequency Security Blade, Return of The Domestic. CDOs coming soon!

### PR DESCRIPTION
**Pull request by Iamgoofball**
## About The Pull Request
dreamseeker_mDcNTGaJ7l
### **The HF blade and it's belt, demonstrated by Joey Earl and Sonny Shuttlecraft.**

https://user-images.githubusercontent.com/4081722/172777848-e9df8d8a-41c9-4a7b-805d-876d96f4f934.mp4

Security Officers are now issued a nerfed High Frequency blade instead of stun batons.
The blade can be quickly drawn from the sheath via alt-click.
Stun Batons can still be ordered via cargo as usual.

Security Officers now start with the Domestic, loaded with Nonlethal rounds. There are six types of rounds available. Nonlethal, TRAC, Iceblox, Geiger, Hyperlethal, and Wealth.

Here's the available ammo types in detail:
Nonlethal: Pure stamina, bouncy. Doesn't wound, full damage. Pacifist compatible.
TRAC: Reduced damage, embeds a tracking chip. Pierces armor. Embeds well.
Iceblox: Reduced damage. Cold. Same as Detective Iceblox.
Hyperlethal: Reduced damage. Heavy wound, Extreme bare wound, almost guaranteed embed.
Geiger: Reduced damage. Irradiation.
Wealth: Damage based on current ID card holdings.

## How This Contributes To The Skyrat Roleplay Experience

Skyrat staff said that they preferred lethal security and disliked stun batons, so here's an alternative to stun batons that's lethal while providing some counter-play to the gun proliferation on the station.

With Guncargo in full swing, security could use something beefier than a Disabler. The playing field is more levelled with the Domestic, which received rave reviews in the previous PR. I've heard that Security teams are resorting to having to do bounties to be able to afford guns to do their job, which isn't great.

## Changelog
:cl: Iamgoofball, security katana idea stolen from Paxilmaniac
add: Adds the Domestic, replacing the starting Disabler on Security Officers.
add: Collect all six ammunition types available via the techweb in Subdermal Implants and Exotic Ammunition!
add: You can get more from cargo crates.
add: Adds the Security High Frequency Blade, replacing the starting Stun Baton in Security Officer lockers.
add: The blade is unable to gib and does less base damage, but scales up the damage if you space out the location of your clicks on the person between hits, just like the Ninja sword.
add: You can get more from cargo crates.
/:cl:

### CODE REVIEW

**Comment by GuiltyNeko** [comment]
The alert locked thing doesn't appear used anywhere, let alone checked in the dme? Let me know if I'm wrong, but it really does look incredibly unused.

**Comment by Iamgoofball** [comment]
Ah, that's setup for the CDOs, I must of committed that by accident.

**Comment by KathrinBailey** [comment]
image

**Comment by Draganfrukts** [comment]
image

**Comment by CandleJaxx** [comment]
ngl based

**Comment by Paxilmaniac** [comment]
There is one small problem with this, don't the HF blades literally gib people if you hit them enough times? Knowing the 'people' that play security there's no doubt it would 'accidentally' happen often.

-> hasn't read the changelog :trollface:

**Comment by Iamgoofball** [comment]
i disabled gibbing on the security variant to prevent that, yeah

**Comment by softcerv** [comment]
you could possibly look into adding a stamina damage mode like the blade has in MGS2 (although I could also understand why you wouldn't want to)

**Comment by Iamgoofball** [comment]
If testing shows they need it, I'll put it in.

**Comment by softcerv** [comment]
you should atomize the domestic change and HF blade change

File: modular_skyrat/modules/goofsec/code/alert_locked_element.dm [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+	alert_level = required_level
+	parent_atom.name = "alert-locked [parent_atom.name]"
+	RegisterSignal(SSsecurity_level, COMSIG_SECURITY_LEVEL_CHANGED, .proc/check_security_level)
+
+/datum/component/alert_locked/proc/check_security_level(datum/source, new_level)
+	SIGNAL_HANDLER
+	if(alert_level > new_level)
+		var/atom/movable/parent_atom = parent
+		parent_atom.balloon_alert_to_viewers("dematerializes due to alert level!")
+		qdel(parent_atom)
```
Comment (softcerv) [inline modular_skyrat/modules/goofsec/code/alert_locked_element.dm:18]:
```suggestion
	if(!alert_level > new_level)
	    return
	var/atom/movable/parent_atom = parent
	parent_atom.balloon_alert_to_viewers("dematerializes due to alert level!")
	qdel(parent_atom)
```

File: modular_skyrat/modules/goofsec/code/domestic.dm [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+
+/obj/projectile/bullet/security_pistol/geiger
+	name = "geiger inhumane bullet"
+	damage = 25
+
+/obj/projectile/bullet/security_pistol/geiger/on_hit(atom/target, blocked = FALSE)
+	. = ..()
+	if (ishuman(target))
+		target.AddComponent(/datum/component/irradiated)
+		radiation_pulse(target, max_range = 1, threshold = RAD_LIGHT_INSULATION)
```
Comment (softcerv) [inline modular_skyrat/modules/goofsec/code/domestic.dm:186]: early return
```suggestion
	if (!ishuman(target))
	        return FALSE
	target.AddComponent(/datum/component/irradiated)
	radiation_pulse(target, max_range = 1, threshold = RAD_LIGHT_INSULATION)
```

**Comment by nikita-chekistov** [comment]
Security's main goal is to arrest, not to turn a criminal into a meat mince while listening to MGR ost. Why would Nanotrasen even give them swords? For what purpose?

My final opinion - cringe.

**Comment by Kush1Push1** [comment]
Unfitting reference to MGR. "Ninja police" seems to me worser than "civilian protection" style.

**Comment by Lizzzard26** [comment]
another goof's cringe

**Comment by Iamgoofball** [comment]
Might as well wait for the testmerge before passing judgement.

**Comment by theselfish** [comment]
Person who hasn't read the PR says...

> Security's main goal is to arrest, not to turn a criminal into a meat mince while listening to MGR ost. Why would Nanotrasen even give them swords? For what purpose?
> 
> My final opinion - cringe.

**Comment by Ashrollen** [comment]
> Security's main goal is to arrest, not to turn a criminal into a meat mince while listening to MGR ost. Why would Nanotrasen even give them swords? For what purpose?
> 
> My final opinion - cringe.

DIdn't read PR. Sees name.Need to troll. Posts cringe himself.Objective complete.Also if we get fleshmind it will totaly explain itself.

**Comment by nikita-chekistov** [comment]
> > Security's main goal is to arrest, not to turn a criminal into a meat mince while listening to MGR ost. Why would Nanotrasen even give them swords? For what purpose?
> > 
> > My final opinion - cringe.
> 
> DIdn't read PR. Sees name.Need to troll. Posts cringe himself.Objective complete.Also if we get fleshmind it will totaly explain itself.

Says goof defender.

**Comment by Iamgoofball** [comment]
unfunny shitposting gets you put on the block list and thus ineligible to shitpost on my PRs

if you're gonna shitpost, be funny about it like @Spc-Dragonfruits

**Comment by Ashrollen** [comment]
> > > Security's main goal is to arrest, not to turn a criminal into a meat mince while listening to MGR ost. Why would Nanotrasen even give them swords? For what purpose?
> > > My final opinion - cringe.
> > 
> > 
> > DIdn't read PR. Sees name.Need to troll. Posts cringe himself.Objective complete.Also if we get fleshmind it will totaly explain itself.
> 
> Says goof defender.

Okay fun fact. Right now security is already lethal enough. Only difference how lethals look. I can kill you same way with stunbatton hitting through rmb while its on. in 4 hits you will lay defenceless fully until i break your skull which will be long and painfull moment with you hearing how slowly you descent in black screen. Disabler? Don't make me laugh people throw it out easily for guncargo guns. In short what looks like sec is not lethal is just facade. I already forgot when sec didn't send someone to deathgasp message while arresting.Even if guy got arrested they probably will have a lot of wounds due to resist.If we continue that discussion i can go for many examples where antags themselves want to go until someone meets horizontal and goes deathgasp.

**Comment by Gandalf2k15** [comment]
okay you know what, fuck it, tm

**Comment by Gandalf2k15** [comment]
Now you can't say we never tried ;)

**Comment by SarmentiCampbell** [comment]
Antags need Murasama

**Comment by Gandalf2k15** [comment]
fr tho the blade is bad but the guns are good, just give them a forceful nightstick or somethinglmao

**Comment by PCPort** [comment]
Oh my god it actually got testmerged. WHEW. Okay.

I'm not going to comment at all on the HF blades, I'm sure lots of folks will have stuff to say about that. But I do recall that the Hyperlethal rounds were perhaps a bit too overpowered. Decapitating in three shots is kinda nutso.

**Comment by Moostard** [comment]
I like it.
People complained about baton being a "1 hit win op weapon". So here is something else.
Security did not lose nonlethal capabilities.
They still got flashes, bangs, domestic got nonlethals loaded and baton crate still exists.
And I like that the research can bring domestic to be able to fight against good firearms without the need for security to literally make factorio setups in the brig for bounties.

**Comment by dankmemerino147** [comment]
domestics good, HF worse than baton but its ok i guess, also afaik **extra magazines for domestics are unobtainable currently**

**Comment by theselfish** [comment]
To relay from my friend who played the round; Fun. Very kinetic, though it's hard to judge them (they only went up against fleshmind.)

**Comment by softcerv** [comment]
The blades and domestics definitely need to be upgraded in price. 600cr for 3 HF blades is absolutely insane.

**Comment by PCPort** [comment]
> The blades and domestics definitely need knocked down a bit in price. 600cr for 3 HF blades is absolutely insane.

DOWN? More like up.

**Comment by softcerv** [comment]
> DOWN? More like up.

mistype, I am tired.

**Comment by ChungusGamer666** [comment]
Nerdbait PR.

**Comment by Iamgoofball** [comment]
okay, changes to make balance wise based on the tests:
1. ID lock the sword crate and domestic crates
2. Make the sword not able to break down structures, unintended behavior
3. Tune down the visual and sound effects

**Comment by Iamgoofball** [comment]
changes to visuals when i get a spriter:
1. make the sword look like an energy nightstick or something

**Comment by ChungusGamer666** [comment]
Weren't you going to stop working for skyrat because you said it sucks on reddit or something

**Comment by Iamgoofball** [comment]
I quit until I felt comfortable working here again. As far as I've heard, Xyel faced consequences for his actions, was held accountable, and most importantly, is working to be better. As such, I feel comfortable coding again.

**Comment by ChungusGamer666** [comment]
> I quit until I felt comfortable working here again. As far as I've heard, Xyel faced consequences for his actions, was held accountable, and most importantly, is working to be better. As such, I feel comfortable coding again.

Yeah i'm 100% sure the host faced consequences for actions in his own server...

**Comment by Iamgoofball** [comment]
He was held accountable by his community for his actions, and afterwards, by all accounts from people I trust, he's made good on his promises to do better by his community and those in it. 

Xyel made a lot of very, very serious mistakes that the SS13 community and the Skyrat community now knows about. He has, however, done his best to make up for those mistakes and do good by people and apologize to those he's wronged, and it would be wrong of me to continue to shun Skyrat as a whole just because of the actions of one person when there are plenty of very good people here.

**Comment by CandleJaxx** [comment]
incredibly based and ngl love the blades

**Comment by Iamgoofball** [comment]
@Funce I pushed the requested changes, do you mind re-TMing it?

**Comment by Whygil** [comment]
PR that just adds a bunch of memes from a guy who tried to sabotage the development by scaring off developers and left the discord server, so he can't even play on Skyrat got approved

**Comment by Iamgoofball** [comment]
The PR is entirely legitimate, not sure why you think otherwise.

**Comment by hugmeorelse** [comment]
> He was held accountable by his community for his actions, and afterwards, by all accounts from people I trust, he's made good on his promises to do better by his community and those in it.
> 
> Xyel made a lot of very, very serious mistakes that the SS13 community and the Skyrat community now knows about. He has, however, done his best to make up for those mistakes and do good by people and apologize to those he's wronged, and it would be wrong of me to continue to shun Skyrat as a whole just because of the actions of one person when there are plenty of very good people here.

Nobody held me accountable in the sense you speak with. At the end of the day, I'm accountable and responsible for my own actions. However, you're overstating your effects on the server along with my effects on the server.

I am currently tolerating your presence in the Github, because I have no desire to be petty. In fact, the other reason I have not blocked you is because I do not believe drama has a place in the Github. So, keep it to Reddit, and keep the drama out of the Github kindly.

**Comment by Higgin** [comment]
Test rounds seemed to go fine to me, changes were needed on the material destruction though. Glad to see that's in. 

Behavior did not change significantly, but that might be something to watch for in the longer run. 

I remember (and having again seen them in action) thinking the lethal Domestic options were very good. I think the damage/wound values on HL and TRAC could be brought down, and the scaling for max damage on Wealth rounds could go much higher. (Alternatively, maybe give HL rounds deeper negative AP to reflect a starker tradeoff for damage against soft targets.) Are you really tapping into the primordial power of the market if you aren't sitting fat on at least 15k? You should have to really be hustling for them to cap out where the cap is right now.

At present they're all pretty dominant options for ease to get and effectiveness vs. other guns. They don't have half the reloading struggle of shotguns (which may also be getting nerfed upstream with the /tg/ move to energy over ballistics,) print more easily and honestly better ammo than the CMGs, are available through free orders or spawn (vs. anything guncargo,) and can be bagged/pocketed. 

I love the versatility and accessibility of them. I think they can be made a more interesting and less dominant choice with some tweaks to their values. Just gating the lethal ammo types with research isn't so much of a challenge on SR because rounds tend to go long enough for it and antags don't tend to attack things like the GOON without an objective (if they're even aware of its significance.) May suggest some updated values after local testing later tonight. 

Even still, all changes seem to be in the right direction. Hope this gets further time as TM.

**Comment by Iamgoofball** [comment]
>Nobody held me accountable in the sense you speak with. At the end of the day, I'm accountable and responsible for my own actions. However, you're overstating your effects on the server along with my effects on the server.

You apologized to your community publicly when they requested an apology and are making things right with those you harmed. That's the definition of accountability and I'm happy to hear that you're doing well.

Regardless, this is offtopic, so let's all try to keep discussion related to the PR itself.

**Comment by Iamgoofball** [comment]
> I think the damage/wound values on HL and TRAC could be brought down, and the scaling for max damage on Wealth rounds could go much higher.

I'll do a pass on the damage numbers.

**Comment by softcerv** [comment]
you should take a look into add basic domestic mags to cargo

**Comment by Iamgoofball** [comment]
They're not officially part of the guncargo lineup.

**Comment by softcerv** [comment]
> They're not officially part of the guncargo lineup.

Even when guncargo wasn't a thing, it was still possible to get ammo for most of the cargo guns, either through separate crates or by the autolathe.

**Comment by Iamgoofball** [comment]
If you want to print security protolathe ammo, then build an ammo workbench or steal the security protolathe board from the Warden.

**Comment by LeonY24** [comment]
metal gear antags when
nanomachines when

**Comment by RazorManager89** [comment]
> nanomachines when

https://github.com/tgstation/tgstation/pull/60473

**Comment by Iamgoofball** [comment]
@LeonY24 virology is basically nanomachines

**Pull request closed without merging**